realm/realm-library/src/main/cpp/object-store/src/shared_realm.hpp

```diff
@@ -148,6 +149,11 @@ namespace realm {
         void init(std::shared_ptr<_impl::RealmCoordinator> coordinator);
         Realm(Config config);
 
+        // FIXME: Java still needs the SharedGroup pointers for some places.
+        //        Remove this after the final clean up.
+        SharedGroup& get_shared_group() { return *m_shared_group; }
```

Comment:
Pushed some new commits which created a some helper functions in OS. Let me know if it looks better, or maybe it looks worse :P